File: atm.c
#include <stdio.h>

int main() {
    int pin, enteredPin, attempts = 3;
    float balance = 1000.0, amount;
    int choice;

    pin = 1234;

    while (attempts > 0) {
        printf("Enter your 4-digit PIN: ");
        scanf("%d", &enteredPin);

        if (enteredPin == pin) {
            while (1) {
                printf("\n--- ATM MENU ---\n");
                printf("1. Check Balance\n");
                printf("2. Deposit Money\n");
                printf("3. Withdraw Money\n");
                printf("4. Exit\n");
                printf("Choose an option: ");
                scanf("%d", &choice);

                switch (choice) {
                    case 1:
                        printf("Current balance: $%.2f\n", balance);
                        break;
                    case 2:
                        printf("Enter amount to deposit: $");
                        scanf("%f", &amount);
                        if (amount > 0) {
                            balance += amount;
                            printf("Deposited: $%.2f\n", amount);
                        } else {
                            printf("Invalid amount.\n");
                        }
                        break;
                    case 3:
                        printf("Enter amount to withdraw: $");
                        scanf("%f", &amount);
                        if (amount > 0 && amount <= balance) {
                            balance -= amount;
                            printf("Withdrawn: $%.2f\n", amount);
                        } else {
                            printf("Invalid or insufficient funds.\n");
                        }
                        break;
                    case 4:
                        printf("Thank you for using the ATM.\n");
                        return 0;
                    default:
                        printf("Invalid choice.\n");
                }
            }
        } else {
            attempts--;
            printf("Incorrect PIN. Attempts left: %d\n", attempts);
        }
    }

    printf("Too many incorrect attempts. Exiting...\n");
    return 0;
}

File: atm.py
class ATM:
    def _init_(self):
        # Simulated user database (username: [PIN, balance])
        self.users = {
            "user1": ["1234", 1000.0],
            "user2": ["5678", 2500.0]
        }
        self.current_user = None

    def login(self):
        print("=== Welcome to Python ATM ===")
        username = input("Enter your username: ")
        if username in self.users:
            pin = input("Enter your PIN: ")
            if pin == self.users[username][0]:
                self.current_user = username
                print(f"\nLogin successful. Welcome, {username}!\n")
                self.menu()
            else:
                print("Invalid PIN.")
        else:
            print("Username not found.")

    def menu(self):
        while True:
            print("1. Check Balance")
            print("2. Deposit")
            print("3. Withdraw")
            print("4. Exit")
            choice = input("Select an option (1-4): ")

            if choice == "1":
                self.check_balance()
            elif choice == "2":
                self.deposit()
            elif choice == "3":
                self.withdraw()
            elif choice == "4":
                print("Thank you for using Python ATM. Goodbye!")
                break
            else:
                print("Invalid option. Please try again.")

    def check_balance(self):
        balance = self.users[self.current_user][1]
        print(f"Your current balance is: ${balance:.2f}\n")

    def deposit(self):
        try:
            amount = float(input("Enter amount to deposit: $"))
            if amount > 0:
                self.users[self.current_user][1] += amount
                print(f"${amount:.2f} deposited successfully.\n")
            else:
                print("Amount must be positive.\n")
        except ValueError:
            print("Invalid input. Please enter a numeric value.\n")

    def withdraw(self):
        try:
            amount = float(input("Enter amount to withdraw: $"))
            balance = self.users[self.current_user][1]
            if 0 < amount <= balance:
                self.users[self.current_user][1] -= amount
                print(f"${amount:.2f} withdrawn successfully.\n")
            else:
                print("Insufficient funds or invalid amount.\n")
        except ValueError:
            print("Invalid input. Please enter a numeric value.\n")


# Run the ATM program
if _name_ == "_main_":
    atm = ATM()
    atm.login()
